pkg/pricer/pricer.go

```diff
@@ -198,17 +239,62 @@ func (s *Pricer) NotifyPriceTable(peer swarm.Address, priceTable []uint64) error
 
 	pricingPeer.lock.Lock()
 	defer pricingPeer.lock.Unlock()
-	s.logger.Debugf("Storing pricetable %v for peer %v", priceTable, peer)
-	err = s.store.Put(peerPriceTableKey(peer), priceTable)
+
+	return s.storePriceTable(peer, priceTable)
+}
+
+func (s *Pricer) NotifyPeerPrice(peer swarm.Address, price uint64, index uint8) error {
+
+	if price == 0 {
+		return nil
+	}
+
+	pricingPeer, err := s.getPricingPeer(peer)
 	if err != nil {
-		return fmt.Errorf("failed to persist pricetable for peer %v: %w", peer, err)
+		return err
 	}
 
-	return nil
+	pricingPeer.lock.Lock()
+	defer pricingPeer.lock.Unlock()
+
+	priceTable := s.peerPriceTable(peer)
+	currentIndexDepth := uint8(len(priceTable)) - 1
+
+	if index <= currentIndexDepth {
+		// Simple case, already have index depth, single value change
+		priceTable[index] = price
+		s.logger.Debugf("Storing updated pricetable %v for peer %v", priceTable, peer)
+		return s.storePriceTable(peer, priceTable)
+	}
+
+	// Complicated case, index is larger than depth of already known table
+	newPriceTable := make([]uint64, index+1)
+
+	// Copy previous content
+	_ = copy(newPriceTable, priceTable)
+
+	// Check how many rows are missing
+	numberOfMissingRows := index - currentIndexDepth
+
+	if numberOfMissingRows > 1 {
```

Comment:
this `if` can probably be removed, just use the `for` loop to enforce this condition